rtt_tf/rtt_tf-component.cpp

```diff
@@ -109,125 +165,148 @@ namespace rtt_tf
-              log(Warning) << "Message received without callerid" << endlog();
-              authority = "no callerid";
-            } else {
-              authority = it->second;
-            }
-            this->setTransform(trans, authority);
-#endif
-          } catch (TransformException& ex) {
-            log(Error) << "Failure to set received transform from "
-              << msg_in.transforms[i].child_frame_id << " to "
-              << msg_in.transforms[i].header.frame_id
-              << " with error: " << ex.what() << endlog();
+          std::map<std::string, std::string>* msg_header_map =
+            msg.__connection_header.get();
+          std::string authority;
+          std::map<std::string, std::string>::iterator it =
+            msg_header_map->find("callerid");
+
+          if (it == msg_header_map->end()) {
+            log(Warning) << "Message received without callerid" << endlog();
+            authority = "no callerid";
+          } else {
+            authority = it->second;
           }
+          this->setTransform(trans, authority);
+#endif
+        } catch (tf2::TransformException& ex) {
+          log(Error) << "Failure to set received transform from "
+            << msg.transforms[i].child_frame_id << " to "
+            << msg.transforms[i].header.frame_id
+            << " with error: " << ex.what() << endlog();
         }
       }
-    } catch (std::exception& ex) {
+    }
+  }
+
+  void RTT_TF::updateHook()
+  {
+    Logger::In(this->getName());
+#ifndef NDEBUG
+    //log(Debug) << "In update" << endlog();
+#endif
+    try
+    {
+      tf2_msgs::TFMessage msg_in;
+
+      internalUpdate(msg_in, port_tf_in);
+      internalUpdate(msg_in, port_tf_static_in);
+    }
+    catch (std::exception& ex)
+    {
       log(Error) << ex.what() << endlog();
     }
   }
 
-  geometry_msgs::TransformStamped RTT_TF::lookupTransform(
+  bool RTT_TF::startHook()
+  {
+    return true;
+  }
+
+  void RTT_TF::stopHook()
+  {
+  }
+
+  void RTT_TF::cleanupHook()
+  {
+  }
+
+  ros::Time RTT_TF::getCommonTime(
       const std::string& target,
-      const std::string& source)
+      const std::string& source) const
   {
-    tf::StampedTransform stamped_tf;
     ros::Time common_time;
-    this->getLatestCommonTime(source, target, common_time,NULL);
-    static_cast<tf::Transformer*>(this)->lookupTransform(target, source, common_time, stamped_tf);
-    geometry_msgs::TransformStamped msg;
-    tf::transformStampedTFToMsg(stamped_tf,msg);
-    return msg;
+
+    tf2::CompactFrameID target_id = _lookupFrameNumber(target);
+    tf2::CompactFrameID source_id = _lookupFrameNumber(source);
+
+    _getLatestCommonTime(source_id, target_id, common_time, NULL);
+
+    return common_time;
+  }
+
+  geometry_msgs::TransformStamped RTT_TF::lookupTransform(
+      const std::string& target,
+      const std::string& source) const
+  {
+    return tf2::BufferCore::lookupTransform(target, source, getCommonTime(target, source));
```

Comment:
Simply pass `ros::Time()` as the last argument (lookup latest available transform) to [lookupTransform()](http://docs.ros.org/kinetic/api/tf2/html/classtf2_1_1BufferCore.html#abcda3c653aba238f8fdd4b903c4bac7b). I think the internal implementation will be more efficient than to lookup the latest common time first.